### PR TITLE
Add settings feature toggle in org settings

### PR DESCRIPTION
**Pull request by malted**
image

### CODE REVIEW

**Review by sarthaktexas — CHANGES_REQUESTED** [review]
Add a background

**Comment by maxwofford** [comment]
I'm fine with this merging without a background just for now because it's a feature flag! We should clarify though, can anyone toggle the feature flag currently?

**Comment by maxwofford** [comment]
2 drafts you can use:

amaziograph_DBC9F70E-9C41-45A4-A444-0EB6BE1EE33B amaziograph


amaziograph_DBC9F70E-9C41-45A4-A444-0EB6BE1EE33B amaziograph

**Comment by maxwofford** [comment]
These aren't relevant to the feature, but still fun patterns:

amaziograph_5FB89CDD-0783-4282-9984-CBF06D26DEBD amaziograph
amaziograph_45819717-5BC1-4088-B6F9-54AD1466CDC7 amaziograph

**Comment by malted** [comment]
image
image

File: app/views/events/edit.html.erb [hunk 1]
```diff
@@ -113,4 +113,30 @@
...
+    <% elsif Flipper.enabled?(:user_permissions_2024_03_09, @event) %>
+      <%# Feature is enabled, let the user opt-out %>
+      <%= link_to "Disable", disable_feature_event_path(feature: :user_permissions_2024_03_09), method: :post, class: "btn bg-accent #{"disabled" unless policy(@event).disable_feature?} tooltipped tooltipped--w", "aria-label": "Only managers can disable feature previews." %>
+    <% else %>
+      <%# Feature is disabled, let the user opt-in %>
+      <%= link_to "Enable", enable_feature_event_path(feature: :user_permissions_2024_03_09), method: :post, class: "btn bg-info #{"disabled" unless policy(@event).enable_feature?} tooltipped tooltipped--w", "aria-label": "Only managers can enable feature previews." %>
+    <% end %>
+  </div>
+  <p>
+    Disallow certain users from performing critical finanical actions.
```
Comment (garyhtou) [inline app/views/events/edit.html.erb]: Can we have a more descriptive copy here? Maybe even link to the `/roles` page

```erb
<%= link_to "learn more about the roles", roles_path %>
```

**Review by garyhtou — COMMENTED** [review]
What happens to member roles when the feature gets disabled? Should they become managers? or silently stay as members? (mentioned in https://github.com/hackclub/hcb/issues/5533)

**Comment by malted** [comment]
image

image


All users and pending invites automatically become managers after the user accepts this dialogue

**Comment by garyhtou** [comment]
@sarthaktexas can you do another review on this.

good to merge after @sarthaktexas's approval

File: app/views/events/edit.html.erb [hunk 2]
```diff
@@ -113,4 +113,40 @@
...
+                    turbo_confirm: non_manager_count > 0 || non_manager_invite_count > 0 ? "⚠️ You have #{non_manager_count} non-manager user#{"s" unless non_manager_count == 1} #{"and #{non_manager_invite_count} non-manager pending user invite#{"s" unless non_manager_invite_count == 1} " if non_manager_invite_count > 0}in your org ⚠️\n\nDisabling the user permissions feature will turn them into #{non_manager_count + non_manager_invite_count > 1 ? "managers" : "a manager"}. Are you sure you want to disable this feature?" : nil,
+                    turbo_method: "post"
+                  } %>
+    <% else %>
+      <%# Feature is disabled, let the user opt-in %>
+      <%= link_to "Enable", enable_feature_event_path(feature: :user_permissions_2024_03_09), method: :post, class: "btn bg-info #{"disabled" unless policy(@event).enable_feature?} tooltipped tooltipped--w", "aria-label": "Only managers can enable feature previews." %>
+    <% end %>
+  </div>
+  <p>
+    Disallow certain users from performing critical finanical actions.
```
Comment (sarthaktexas) [inline app/views/events/edit.html.erb]:
```suggestion
    Disallow certain users from performing critical financial actions!
```

File: app/assets/stylesheets/components/_features.scss [hunk 3]
```diff
@@ -136,6 +136,31 @@ html[data-dark='true'] .feature-anonymous-donations {
...
+    position: absolute;
+    background-image: #{image-url('feature-user-perms-dark.png')};
+    background-size: 5em;
+    opacity: 0.1;
+  }
+
+  html[data-dark='true'] & {
+    background-color: $black;
+    &::before {
+      opacity: 0.8;
```
Comment (sarthaktexas) [inline app/assets/stylesheets/components/_features.scss]:
```suggestion
      opacity: 0.3;
```

**Review by sarthaktexas — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/components/_features.scss (hunk 3 above)
Comment (sarthaktexas) [reply app/assets/stylesheets/components/_features.scss]: Your choice if you want to keep this or not:
Screenshot 2024-03-18 at 2 15 16 PM

Comment (malted) [reply app/assets/stylesheets/components/_features.scss]: Looks good; cheers!

**Comment by malted** [comment]
I work at a bank and spelled financial wrong 🎀

**Review by sarthaktexas — APPROVED** [review]

**Pull request merged**